providers/mimetypes.rb

```diff
@@ -0,0 +1,68 @@
+#
+## Cookbook Name:: gecos-ws-mgmt
+## Provider:: pdf_viewer
+##
+## Copyright 2013, Junta de Andalucia
+## http://www.juntadeandalucia.es/
+##
+## All rights reserved - EUPL License V 1.1
+## http://www.osor.eu/eupl
+##
+
+require 'etc'
+require 'chef/mixin/shell_out'
+include Chef::Mixin::ShellOut
+
+action :setup do
+	begin
+	  if new_resource.support_os.include?($gecos_os)
+    	package "xdg-utils" do
+       action :nothing
+      end.run_action(:install)
+
+      users = new_resource.users
+
+      users.each_key do |user_key|
+        nameuser = user_key
+        username = nameuser.gsub('###','.')
+        user = users[user_key]
+				gid = Etc.getpwnam(username).gid
+				xdg_code = ''
+
+				directory "/home/#{username}/.local/share/applications" do
+					owner username
+					group gid
+					recursive true
+				end
+
+				Chef::Log.debug("mimetypes.rb - Users: #{user}")
+
+				user.mimetyperelationship.each do |assoc|
```

Comment:
Missed check before acting. An xdg-mime query default mimetype will give you the current association.